### ISSUE TITLE
Change license from AGPL3 to GPL3

### ISSUE THREAD

**Issue by magicus** [Wynntils/Wynntils#1969]
## TL;DR

If you have been pinged here, you have contributed code to Artemis. If you agree with changing license to GPL3 for your contributions to Artemis, **please type "I agree" in a comment below**.

## Background

Due to historic reasons, Artemis was created with the same license as Legacy, the [GNU Affero General Public License v3 (AGPL3)](https://www.gnu.org/licenses/agpl-3.0.en.html). This is a slightly modified version of GPL3, which prohibits public use on a server without releasing the source code. This extra clause is not relevant for Artemis, but it will still make the source code license incompatible with GPL3, a much more common license for Minecraft mods.

I propose that we change the Artemis license from AGPL3 to the normal [GNU General Public License v3 (GPL3)](https://www.gnu.org/licenses/gpl-3.0.html).

Since the code currently present in the Artemis repo was contributed under the AGPL3 license, all contributors must agree to a change. If an author does not agree, or does not reply, we might be forced to remove those commits that said author have provided.

Note that this change do not apply to other Wynntils software, such as Hades and Athena that are running on the Wynntils servers. They will keep using AGPL.

## Process

If you have been pinged in this message, please type "I agree" (or, "I do not agree") in a comment below. An Artemis maintainer will then update the list below based on the responses.

If and when all authors have agreed, we will re-license the project. If a few individual authors do not agree or does not reply, the Artemis development team might decide to remove those contributions and still go ahead with the re-license, keeping only code from those who agree.

## Contributors

Here is a complete list of all contributors to the Artemis repo, [as provided by Github](https://github.com/Wynntils/Artemis/graphs/contributors). 

**Legend:** ❓ = has not replied,     ✅ = has accepted,     ❌ = has not accepted

| User  | Commits  | Approval  |
|---|---|---|
| @kristofbolyai | [875 commits](https://github.com/Wynntils/Artemis/commits?author=kristofbolyai) | ✅ |
| @magicus | [639 commits](https://github.com/Wynntils/Artemis/commits?author=magicus) | ✅ |
| @Incompleteusern | [169 commits](https://github.com/Wynntils/Artemis/commits?author=Incompleteusern) | ❓ |
| @DonkeyBlaster | [95 commits](https://github.com/Wynntils/Artemis/commits?author=DonkeyBlaster) | ✅ |
| @P0keDev | [69 commits](https://github.com/Wynntils/Artemis/commits?author=P0keDev) | ❓ |
| @DevScyu | [36 commits](https://github.com/Wynntils/Artemis/commits?author=DevScyu) | ❓ |
| @san7890 | [27 commits](https://github.com/Wynntils/Artemis/commits?author=san7890) | ❓ |
| @Rafii2198 | [20 commits](https://github.com/Wynntils/Artemis/commits?author=Rafii2198) | ✅ |
| @Mcrtin | [20 commits](https://github.com/Wynntils/Artemis/commits?author=Mcrtin) | ✅ |
| @HighCrit | [16 commits](https://github.com/Wynntils/Artemis/commits?author=HighCrit) | ❓ |
| @JamieCallan117 | [15 commits](https://github.com/Wynntils/Artemis/commits?author=JamieCallan117) | ✅ |
| @HalcyonSuoh | [9 commits](https://github.com/Wynntils/Artemis/commits?author=HalcyonSuoh) | ❓ |
| @TomaSajt | [6 commits](https://github.com/Wynntils/Artemis/commits?author=TomaSajt) | ✅ |
| @byBackfish | [6 commits](https://github.com/Wynntils/Artemis/commits?author=byBackfish) | ✅ |
| @jh-devv | [5 commits](https://github.com/Wynntils/Artemis/commits?author=jh-devv) | ✅ |
| @re-vc | [4 commits](https://github.com/Wynntils/Artemis/commits?author=re-vc) | ✅ |
| @joshieman06 | [4 commits](https://github.com/Wynntils/Artemis/commits?author=joshieman06) | ✅ |
| @DalwynWasTaken | [4 commits](https://github.com/Wynntils/Artemis/commits?author=DalwynWasTaken) | ❓ |
| @mbnkwm | [3 commits](https://github.com/Wynntils/Artemis/commits?author=mbnkwm) | ✅ |
| @ShadowShift | [3 commits](https://github.com/Wynntils/Artemis/commits?author=ShadowShift) | ❓ |
| @ekgame | [3 commits](https://github.com/Wynntils/Artemis/commits?author=ekgame) | ✅ |
| @coehlrich | [3 commits](https://github.com/Wynntils/Artemis/commits?author=coehlrich) | ✅ |
| @McPlayHD | [2 commits](https://github.com/Wynntils/Artemis/commits?author=McPlayHD) | ✅ |
| @CringeAura | [2 commits](https://github.com/Wynntils/Artemis/commits?author=CringeAura) | ❓ |
| @Zatzou | [2 commits](https://github.com/Wynntils/Artemis/commits?author=Zatzou) | ❓ |
| @Jimmy-sheep | [2 commits](https://github.com/Wynntils/Artemis/commits?author=Jimmy-sheep) | ✅ |
| @ENORMOUZ | [2 commits](https://github.com/Wynntils/Artemis/commits?author=ENORMOUZ) | ✅ |
| @bricefrisco | [2 commits](https://github.com/Wynntils/Artemis/commits?author=bricefrisco) | ✅ |
| @YouSoPugly | [2 commits](https://github.com/Wynntils/Artemis/commits?author=YouSoPugly) | ❓ |
| @xCykrix | [1 commit](https://github.com/Wynntils/Artemis/commits?author=xCykrix) | ✅ |
| @atiedebee | [1 commit](https://github.com/Wynntils/Artemis/commits?author=atiedebee) | ❓ |
| @TUsama | [1 commit](https://github.com/Wynntils/Artemis/commits?author=TUsama) | ✅ |
| @Ldude162 | [1 commit](https://github.com/Wynntils/Artemis/commits?author=Ldude162) | ✅ |
| @Kepler-17c | [1 commit](https://github.com/Wynntils/Artemis/commits?author=Kepler-17c) | ❓ |
| @cmkohnen | [1 commit](https://github.com/Wynntils/Artemis/commits?author=cmkohnen) | ✅ |
| @obj-obj | [1 commit](https://github.com/Wynntils/Artemis/commits?author=obj-obj) | ❓ |
| @iwei20 | [1 commit](https://github.com/Wynntils/Artemis/commits?author=iwei20) | ❓ |
| @DevChromium | [1 commit](https://github.com/Wynntils/Artemis/commits?author=DevChromium) | ✅ |
| @KaspianDev | [1 commit](https://github.com/Wynntils/Artemis/commits?author=KaspianDev) | ✅ |

**Comment by magicus** [comment]
I agree

**Comment by McPlayHD** [comment]
I agree

**Comment by ekgame** [comment]
I agree

**Comment by mbnkwm** [comment]
I agree

**Comment by kristofbolyai** [comment]
I agree

**Comment by byBackfish** [comment]
I agree

**Comment by re-vc** [comment]
I agree

mbnkwm ***@***.***> schrieb am Mi. 9. Aug. 2023 um 11:26:

**Comment by sofiedotcafe** [comment]
I agree

**Comment by coehlrich** [comment]
I agree

**Comment by TUsama** [comment]
I agree

**Comment by DevChromium** [comment]
I agree

**Comment by Mcrtin** [comment]
I agree

**Comment by Rafii2198** [comment]
I agree

**Comment by ENORMOUZ** [comment]
I agree

**Comment by ShadowCat117** [comment]
I agree

**Comment by xCykrix** [comment]
I agree

**Comment by KaspianDev** [comment]
I agree since the clause is irrelevant to this type of software.

**Comment by TomaSajt** [comment]
I agree

**Comment by bricefrisco** [comment]
I agree

**Comment by cmkohnen** [comment]
I agree

**Comment by Jimmy-sheep** [comment]
I agree, thank you.

**Comment by joshieman06** [comment]
I agree

**Comment by iwei20** [comment]
Are we sure the GPL can apply here?
- https://www.gnu.org/licenses/gpl-faq.en.html#GPLPluginsInNF - Minecraft being the "nonfree program", Wynntils being the "GPL covered plug-in"
- https://blog.modrinth.com/p/licensing-guide - section "Copyleft license" suggests use of LGPL-3.0 over GPL-3.0

**Comment by derivated** [comment]
I agree

**Comment by ryanzhoudev** [comment]
I agree

**Comment by magicus** [comment]
> Are we sure the GPL can apply here?

@iwei20 That is a relevant question. I just wanted to get rid of the Affero part, and did not stop to think if GPL would be suitable.

It does indeed seem that GPL is not a suitable license for Minecraft mods, and that LGPL is required. I'll confer with the rest of the Wynntils Dev Team about how to proceed. 

Thanks for bringing this to our attention!

**Comment by KaspianDev** [comment]
GPL is totally fine for plugins/mods, even Bukkit API uses GPL.

**Comment by sofiedotcafe** [comment]
> > Are we sure the GPL can apply here?
> 
> @iwei20 That is a relevant question. I just wanted to get rid of the Affero part, and did not stop to think if GPL would be suitable.
> 
> It does indeed seem that GPL is not a suitable license for Minecraft mods, and that LGPL is required. I'll confer with the rest of the Wynntils Dev Team about how to proceed. 
> 
> Thanks for bringing this to our attention!

I'd say GPL is suitable

**Comment by magicus** [comment]
> GPL is totally fine for plugins/mods, even Bukkit API uses GPL.

At the very least, that is apparently subject to legal debate and uncertainty. Using LGPL would remove any uncertainty about the legality.

**Comment by magicus** [comment]
This is superseded by https://github.com/Wynntils/Artemis/issues/1970
